xamarin/Xamarin.Forms
# [Shell] Fix back button behavior so that it only applies the property you've set opposed to replacing the entire thing

### Description of Change ###
This changes BackButtonBehavior so that it only overrides the properties you specify and not the entire thing. This allows users to attach a command or change the icon on the BackButton without totally breaking everything

### Issues Resolved ### 
- fixes #5744

### Platforms Affected ### 
- iOS
- Android

### Testing Procedure ###
- I added a UI test to make sure command works
- There are buttons that you can use to set all the back button behaviors. Just click on different combinations to make sure it all works how you would expect it to

### PR Checklist ###
